index.js

```diff
@@ -135,32 +119,30 @@ module.exports = fp(
         }
 
         let stats;
-
-        if (customDatsClient) {
+        // If client is a custom client or is not an options object
+        if (client instanceof Client || isCustomClient(client)) {
```

Comment:
What do you think if we refactor this check, in a function as `getClientNotImplementedMethods`? and if the output is not empty, we can throw with the methods not implemented.

Reply:
Do you mean to change `isCustomClient`?. The logic here is doing a quick check to see if the object could be a custom client that is not a Dats instance (we could just support Dats instances maybe?), so I didn't want to throw at this point, because if this check fails then I am going to treat `client` as config options to create a new Dats client. Inside the if block we need top throw indeed.

Reply:
Ok, now I have seen that client is also used as a configuration option. Now, I understand the check :+1:

Reply:
As you suggested, I refactor `isCustomClient`